pyhive/hive.py

```diff
@@ -213,17 +211,35 @@ def __init__(
                         password = 'x'
 
                 def sasl_factory():
-                    sasl_client = sasl.Client()
-                    sasl_client.setAttr('host', host)
-                    if sasl_auth == 'GSSAPI':
-                        sasl_client.setAttr('service', kerberos_service_name)
-                    elif sasl_auth == 'PLAIN':
-                        sasl_client.setAttr('username', username)
-                        sasl_client.setAttr('password', password)
-                    else:
-                        raise AssertionError
-                    sasl_client.init()
-                    return sasl_client
+                    try:
+                        import sasl
+                        # The sasl library is available
+                        sasl_client = sasl.Client()
```

Comment:
let's move it into a separate function to keep cleaner
e.g. 
```
def get_sasl_client 
...
```

and 
```def get_pure_sasl_client 
...
```

Reply:
Addressed. Please have a look.